projects/core/src/lib/map/map-layer.component.ts

```diff
@@ -311,12 +320,16 @@ export class OlRasterLayerComponent
 
         this.source.setTileLoadFunction((tile, src) => {
             const client = new XMLHttpRequest();
-
-            const sub = this.projectService.createQueryAbortStream().subscribe(() => {
-                (tile as any).getImage().src = 'assets/images/empty_tile.png';
-                client.abort();
-                sub.unsubscribe();
-                this.source.dispatchEvent('tileloaderror');
+            const tileZoomLevel: number = (tile as any).getTileCoord()[0];
+            const tileResolution = this.mapService.getView().getResolutionForZoom(tileZoomLevel);
+
+            const sub = this.projectService.createQueryAbortStream().subscribe((newResolution) => {
+                if (newResolution !== tileResolution) {
+                    (tile as any).setState(TileState.EMPTY);
+                    client.abort();
+                    sub.unsubscribe();
```

Comment:
Why do you unsubscribe from yourself?

1. You'll get blind from this.
2. Just make sure that this stream finishes after the first `onNext`. You can achieve this for instance with `first()`.

Reply:
it's no longer the case that we only need the first event. It may be that the first event is just the delayed zoom event. We need to keep the subscription until we either decide to abort, or the request finishes. That's why I have a unsubscribe here and one in the onload function

Reply:
As discussed, you can refactor the `createQueryAbortStream` method to incorporate the zoom level as parameter.
Then, you only need to wait for it to push the one and only abort event.

Reply:
you mean the resolution and not the zoom level, right? Shall I also include the map extent then? Otherwise we still have to listen for multiple events.

Reply:
Both, you should make a queryAbortStream that actually fires when the query should be aborted.

Reply:
4f52ff455e47336a8b782b01bf7c826bcf07c9a4